README.md

```diff
@@ -0,0 +1,163 @@
+
+![Domains](./docs/domains.png?raw=true)
+
+
+## Roles
+
+* **Initiator** - person initiating DAI transfer by calling `initiateWormhole` . They can optionally specify Operator and Receiver 
+* **Operator** - person (or specified third party) responsible for initiating minting process on destination domain by providing (in the fast path) Oracle attestations. Can call `requestMint` on `WormholeOracleAuth`
+* **Receiver** - person receiving minted DAI on a destination domain
+
+## DAI Wormhole L2 → L1 (aka fast withdrawals) 
+
+![FastWithdrawal](./docs/fw.png?raw=true)
+
+
+### Normal (fast) path
+
+To fast withdraw DAI from L2, user:
+
+* Calls `l2bridge.initiateWormhole()` - this burns DAI on L2 and sends `finalizeRegisterWormhole()` L2 -> L1 message to withdraw DAI from L2 bridge. This message, in normal cicumstances, will never be relayed and it will eventually expire in L1 message queue
+* Waits for withdrawal attestations to be available and obtains them via Oracle API
+* Calls `WormholeOracleAuth.requestMint(WormholeGUID wormholeGUID, bytes signatures, uint256 maxFee)` which will:
+  * Check if `sender` is `operator` 
+  *   Check if enough valid attestations (sigs) are provided
+  *   Call `WormholeJoin.requestMint(wormholeGUID, maxfee)` which will
+        * Check if this wormhole hasn't been used before
+        * Check if the debt ceiling hasn't been reached
+        * Check the current fee via `WormholeFees`
+        * `vat.slip`, `vat.frob`, `daiJoin.exit`
+
+### Settlement
+
+Settlement process moves DAI from L1 Bridge to WormholeJoin to clear the debt that accumulates there. It is triggered by keepers.
+
+* On L2 keeper calls `l2bridge.flush()`
+* L2 -> L1 message `finalizeFlush()` is sent to `L1Bridge` and relayed by a keeper
+* `L1Bridge` upon receiving `finalizeFlush()` calls `WormholeRouter.settle()` which will
+    * Transfer `DAI` from bridges' escrow to `WormholeJoin`
+    * Call `WormholeJoin.settle()` which will use transfered DAI to clear any outstanding debt by calling `daiJoin.join`, `vat.frob`, `vat.slip`
+
+### Slow (emergency) path
+
+If attestations cannot be obtained (Oracles down or censoring), user needs to wait so that L2 message is confirmed on L1 (on Optimistic Rollups that typically is 7 days, on zkRollups it can be anything between few hours to a day). Once L2->L1 message can be relayed, user:
+
+* Relays `finalizeRegisterWormhole()`  message to `L1Bridge`
+* `L1Bridge` upon receiving `finalizeRegisterWormhole()` will call `requestMint()` on `WormholeRouter` which will:
+    * Call `WormholeJoin.requestMint(wormholeGUID, maxfee)` which will
+        * Check if this wormhole hasn't been used before
+        * Check if the debt ceiling hasn't been reached
+        * Check the current fee via `WormholeFees`
+        * `vat.slip`, `vat.frob`, `daiJoin.exit`
+
+## DAI Wormhole L2→L2
+
+![Wormhole](./docs/l2.png?raw=true)
```

Comment:
The red arrow from L2-B Bridge to WormholeJoin has a typo: `requestMin()` instead of `requestMint()`.

Reply:
fixed